### DISCUSSION TITLE
.env.local not available on deployed app

### DISCUSSION THREAD

**Discussion by terjeee** [vercel/community#3032] [Help] [answered]
### Summary

I'm trying to get NextAuth with the Google Provider working. The application works in my local dev environment, but when I try to access it on the WWW through Vercel's deployment, I get this error in the server logs: [next-auth][error][NO_SECRET]

The issue is that the environment variable "process.env.NEXTAUTH_SECRET" isnt available (see picture). When I change "secret" to a string rather than pointing at it through process.env the application proceeds, then throws an error as it is missing the followingenvironment variable "process.env.GOOGLE_CLIENT_ID"

How can I get the app working in production without exposing keys from .env.local that shouldnt be public?

![image](https://github.com/vercel/community/assets/96795863/caf70120-3e99-4d38-9c73-5601c2e1dde9)

### Steps to Reproduce

The app works in local development, but not when hosted on Vercel through the dashboard (vercel.com/dashboard) because .env.local variables are undefined.

**Comment by amyegan** [comment, marked as answer]
Did you add the [environment variables](https://vercel.com/docs/concepts/projects/environment-variables) to your project settings for preview and production deployments? If not, you can import or copy/paste the .env file to quickly upload them all at once.

**Reply by terjeee** [reply]
Thanks Amy! Have a great day :)
